src/systems/ec/implementations/vanilla/packages/ec-component-form-checkbox/ec-component-form-checkbox.scss

```diff
@@ -0,0 +1,103 @@
+/**
+ * Checkbox component
+ * @define checkbox
+ */
+
+// Import base
+@import '@ecl/ec-base/ec-base';
+
+@mixin ecl-checkbox(
+  $thickness-base: 1px,
+  $thickness-max: 3px,
+  $border-color: $ecl-color-grey-25,
+  $border-color-focus: $ecl-color-yellow-100,
+  $border-color-checked: $ecl-color-blue-100,
+  $fake-checkbox-size: 0.875rem
+) {
+  .ecl-checkbox {
+    align-items: center;
+    display: flex;
+    margin: 0;
+    position: relative;
+  }
+
+  .ecl-checkbox__input {
+    margin-left: 3px;
+    opacity: 0;
+    padding: 0;
+    position: absolute;
+    z-index: 1;
+  }
+
+  .ecl-checkbox__label {
+    color: $ecl-color-grey-100;
+    font: $ecl-font-m;
+  }
+
+  .ecl-checkbox__label--hidden {
+    clip: auto;
+    height: auto;
+    margin: 0;
+    position: static;
+    text-indent: -9999px;
+    width: auto;
+  }
+
+  .ecl-checkbox__box {
+    border: $thickness-base solid $border-color;
+    display: inline-flex;
+    margin-right: $ecl-spacing-s;
+    padding: calc(#{$thickness-max} - #{$thickness-base});
+
+    .ecl-checkbox__label--hidden & {
+      margin: 0;
+    }
+  }
+
+  .ecl-checkbox__icon {
+    fill: $ecl-color-white-100;
+    height: $fake-checkbox-size;
+    visibility: hidden;
+    width: $fake-checkbox-size;
+  }
+
+  .ecl-checkbox--invalid .ecl-checkbox__box {
+    border-color: $ecl-color-error;
+  }
+
+  .ecl-checkbox__input:checked {
+    + label .ecl-checkbox__box {
+      background: $border-color-checked;
+      border-color: $border-color-checked;
+    }
+
+    + label .ecl-checkbox__icon {
+      visibility: visible;
+    }
+  }
+
+  .ecl-checkbox__input:focus,
+  .ecl-checkbox__input:active {
+    + label > .ecl-checkbox__box {
+      border: $thickness-max solid $border-color-focus;
```

Comment:
for the focus, we should use `outline` instead of `border`

Reply:
Ok, it has to be said that all the other form elements defined in the ECL v2 are actually using a border, though.
About the input i now used the technique you suggested, but this is now preventing the checkbox from working in case the label is missing the "for" attribute, while the previous solution was a little bit more flexible, from this point of view.

Reply:
You're completely right about the other components, we should fix them as well!
Thank you for reporting it, really appreciated 👍 

The `for` attribute is mandatory since the `input` is outside the `label`, and this is a good thing. The fact that it was working without it just because there's was some obscure hidden input above the icon was a dangerous side effect (am I exaggerating? 😄). Now, the `for` attribute also semantically links the label with the input. For screen readers, this is an absolute necessity (unless you have an `aria-label` but this is another topic 😆)